src/libraries/System.Data.OleDb/src/OleDbDataReader.cs

```diff
@@ -470,7 +470,7 @@ private void BuildSchemaTableInfo(object handle, bool filterITypeInfo, bool filt
             }
 
             OleDbHResult hr;
-            IntPtr columnCount = ADP.PtrZero; // column count
+            nint columnCount = ADP.PtrZero; // column count
```

Comment:
```suggestion
            nint columnCount = 0; // column count
```
Also, it would be best to delete `ADP.PtrZero` and replace all instances with just 0.